### PR TITLE
testing performance and minor refactoring

### PR DESCRIPTION
**Pull request by gcolvin**
Test speed with and without recent improvements, moved Pawel's new exp() code to VMOpt.cpp, removed obscuring uses of auto, other housekeeping.

### CODE REVIEW

File: libevm/VMConfig.h [hunk 1]
```diff
@@ -33,17 +33,17 @@ namespace eth
...
 	#define EVM_SWITCH_DISPATCH
 #endif
 
-#if false
+#if true
 	#define EVM_REPLACE_CONST_JUMP
 #endif
 
-#if false
+#if true
```
Comment (chriseth) [inline libevm/VMConfig.h]: Are these leftovers?

File: libevm/VMConfig.h [hunk 2]
```diff
@@ -54,7 +54,9 @@ namespace eth
 
 
 // set this to 2, 1, or 0 for more, less, or no tracing to cerr
-#define EVM_TRACE 0
+#ifndef EVM_TRACE
```
Comment (chfast) [inline libevm/VMConfig.h]: In case a macro is not defined `#if` should replace it with `0`. Please remove it, unless any compiler complains about it.

**Comment by chfast** [comment]
However, clang complains about
```
/Users/travis/build/ethereum/cpp-ethereum/libevm/VM.cpp:738:19: error: unsequenced modification and access to 'm_pc' [-Werror,-Wunsequenced]
                        m_pc += m_code[++m_pc];
                             ~~        ^
```
And I agree.

File: libevm/VMConfig.h (hunk 2 above)
Comment (gcolvin) [inline libevm/VMConfig.h]: You are right.

**Comment by gcolvin** [comment]
Yes, clang is right.  I keep fixing that line, and git keeps merging it back in.

**Comment by gcolvin** [comment]
As for auto - I don't like it for scalars where I do care and should know what they are.  As compared to unthinkable iterator types that just work somehow.

File: libevm/VMConfig.h (hunk 1 above)
Comment (gcolvin) [inline libevm/VMConfig.h]: Leftovers?  I've done more testing so I'm turning them on.

**Comment by gcolvin** [comment]
@chriseth @chfast  Can someone push Merge button?

**Comment by gcolvin** [comment]
Thanks, @chriseth

**Pull request merged**